zapcore/field.go

```diff
@@ -199,3 +199,25 @@ func addFields(enc ObjectEncoder, fields []Field) {
 		fields[i].AddTo(enc)
 	}
 }
+
+func encodeStringer(key string, stringer interface{}, enc ObjectEncoder) {
+	defer catchPanic(key, enc)
+	enc.AddString(key, stringer.(fmt.Stringer).String())
+}
+
+func catchPanic(key string, enc ObjectEncoder) {
+	if err := recover(); err != nil {
+		var msg string
+
+		switch v := err.(type) {
```

Comment:
can't reply to the conversation about matching `fmt`, but I like the idea of `PANIC=` to make it explicit that this was from a panic (rather than a standard returned error).

I like their use of `<nil>` as well, though that seems like it needs string matching, which I'm not a fan of. @abhinav thoughts?

Reply:
The `PANIC=` prefix sounds nice. It helps differentiate between `panic("foo")`
and `return errors.New("foo")` from an error-able encode call.

I'm not a fan of the `nil` check. `PANIC=runtime error: invalid memory address or nil pointer dereference` is enough information for the user to know what was nil.

Reply:
`fmt` performs a `nil` check via reflection at this stage. Given that this code path should be rather cold it is something to consider.

Reply:
```go
if v := recover(); v != nil {
  err = errors.New(fmt.Sprintf("PANIC=%v", v))
}
```

This has a nice simplicity to it.